File: mapir-android-service-sdk/src/main/java/ir/map/servicesdk/response/StaticMapResponse.java
package ir.map.servicesdk.response;

import android.graphics.Bitmap;

import java.io.InputStream;

import ir.map.servicesdk.model.base.MapirResponse;

public class StaticMapResponse extends MapirResponse {

    private InputStream data;
    private Bitmap bitmap;

    public StaticMapResponse(InputStream inputStream, Bitmap bitmap) {
        this.data = inputStream;
        this.bitmap = bitmap;
    }

    public static StaticMapResponse createStaticMapResponse(InputStream data) {
        return new StaticMapResponse(data, null);
    }

    public InputStream getData() {
        return data;
    }

    public Bitmap getBitmapStaticMap() {
        return bitmap;
    }
}

File: mapir-android-service-sdk-ktx/src/main/java/ir/map/servicesdk/response/StaticMapResponse.kt
package ir.map.servicesdk.response

import android.graphics.Bitmap
import ir.map.servicesdk.model.base.MapirResponse
import java.io.InputStream

class StaticMapResponse(val data: InputStream, val bitmapStaticMap: Bitmap? = null) : MapirResponse() {
    companion object {
        fun createStaticMapResponse(data: InputStream) = StaticMapResponse(data, null)
    }
}